ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -157,6 +177,58 @@ public void run() {
         new PartitionSelectionHelper(partitionMap.values(), clusterMapConfig.clusterMapDatacenterName);
   }
 
+  /**
+   * Initialize HelixManager in local datacenter and complete subscription of HelixPropertyStore to listen for
+   * PartitionOverride zNode.
+   * @param dataCenterToZkAddress the map mapping each datacenter to its corresponding ZkAddress.
+   * @param instanceName the String representation of the instance associated with this manager.
+   * @param helixFactory the factory class to construct and get a reference to a {@link HelixManager}.
+   * @return the HelixManager of local datacenter
+   */
+  private HelixManager initializeHelixManagerAndPropertyStoreInLocalDC(Map<String, DcZkInfo> dataCenterToZkAddress,
+      String instanceName, HelixFactory helixFactory) {
+    DcZkInfo dcZkInfo = dataCenterToZkAddress.get(clusterMapConfig.clusterMapDatacenterName);
+    String zkConnectStr = dcZkInfo.getZkConnectStr();
+    HelixManager manager = null;
+    try {
+      manager = helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.SPECTATOR, zkConnectStr);
+      helixPropertyStore = manager.getHelixPropertyStore();
+      logger.info("Connecting to Helix manager in local datacenter at {}", zkConnectStr);
+      manager.connect();
+      logger.info("Established connection to Helix manager in local datacenter at {}", zkConnectStr);
+    } catch (Exception e) {
+      initializationException.compareAndSet(null, e);
```

Comment:
should this be an immediate failure? This is not being done in its own thread - it's in the main thread, right?